crates/wasmtime/src/runtime/externals/table.rs

```diff
@@ -94,7 +94,8 @@ impl Table {
     /// # }
     /// ```
     pub fn new(mut store: impl AsContextMut, ty: TableType, init: Ref) -> Result<Table> {
-        Table::_new(store.as_context_mut().0, ty, init)
+        vm::one_poll(Table::_new(store.as_context_mut().0, ty, init))
+            .expect("must use `new_async` when async resource limiters are in use")
```

Comment:
Should we not assert that this is a non-async config before the `.expect(..)` to catch mismatches a little earlier?

Reply:
Oh this is actually replicating the documented behavior where this works with `async_support` so long as an async resource limiter isn't used. In that sense to avoid breaking the semantics here `one_poll` is what's required as opposed to an `assert!(!async_support)` plus `vm::assert_ready`.